modules/opengl/src/shader/shader.cpp

```diff
@@ -97,11 +106,11 @@ Shader::Shader(const Shader &rhs) : id_{glCreateProgram()}, warningLevel_{rhs.wa
     ShaderManager::getPtr()->registerShader(this);
 }
 
-Shader::Shader(Shader &&rhs) : id_{rhs.id_}, ready_(rhs.ready_), warningLevel_{rhs.warningLevel_} {
+Shader::Shader(Shader &&rhs) : ready_(rhs.ready_), warningLevel_{rhs.warningLevel_} {
 
     ShaderManager::getPtr()->unregisterShader(&rhs);
-
-    rhs.id_ = 0;
+    prog_ = rhs.prog_;
```

Comment:
there is a leak here! `this->prog_` is already initialized and has a program. Either you delete that program first or swap `prog_` with `rhs.prog_`

Reply:
This is the same behavior as before. I assumed that  `ShaderManager::getPtr()->unregisterShader(&rhs);` above deleted the program. Will double check

Reply:
So I checked and it does not seem remove the program in that way. Will adjust the ScopedShaderProgram

Reply:
Fixed it